extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreIntegrationTest.java

```diff
@@ -128,4 +128,36 @@ public void testRemove() throws Exception {
 
         assertThat(jdbcRowsTable(tableName)).isEmpty();
     }
+
+    @Test
+    public void testDynamicExternalDataStoreConfig() throws Exception {
+        String randomTableName = randomTableName();
+
+        createTable(randomTableName);
+        assertThat(jdbcRowsTable(randomTableName)).isEmpty();
+
+        HazelcastInstance client = client();
+
+        client.getConfig().addExternalDataStoreConfig(
+                new ExternalDataStoreConfig("dynamically-added-datastore")
+                        .setClassName(JdbcDataStoreFactory.class.getName())
+                        .setProperty("jdbcUrl", dbConnectionUrl)
+        );
+
+        MapStoreConfig mapStoreConfig = new MapStoreConfig()
+                .setClassName(GenericMapStore.class.getName())
+                .setProperty(EXTERNAL_REF_ID_PROPERTY, "dynamically-added-datastore")
```

Comment:
Idea for the future (not  for this PR): if EXTERNAL_REF_ID_PROPERTY is a commonly known property, then create setter for this particular property. You would then have `setExternalRefId("dynamically-added-datastore")`. Improvement idea for the future ;)